### PR TITLE
Sort schemes in Xcode

### PR DESCRIPTION
**Pull request by David-Lam**
Fixes https://github.com/MobileNativeFoundation/rules_xcodeproj/issues/1970

This pr generate `PROJECT.xcodeproj/xcuserdata/$USER.xcuserdatad/xcschememanagement.plist` with `orderHint` set to the order of manual specified scheme (user listed order) > auto-generated top level scheme (alphabetic) > auto-generated non-top level scheme (alphabetic).

New scheme order with rules_ios example:
Screenshot 2023-05-01 at 12 45 16 PM

### CODE REVIEW

**Comment by yongjincho92** [comment]
@David-Lam you have to sign your git commits to pass DCO

**Review by yongjincho92 — APPROVED** [review]
approving for running checks

**Review by yongjincho92 — DISMISSED** [review]
approving again for running tests

**Review by brentleyjones — DISMISSED** [review]
Approve for CI run

File: xcodeproj/internal/templates/installer.sh [hunk 1]
```diff
@@ -147,7 +147,6 @@ rsync \
   --chmod=u+w,F-x \
   --exclude=project.xcworkspace \
   --exclude=rules_xcodeproj/bazel \
-  --exclude=xcuserdata \
```
Comment (erikkerber) [inline xcodeproj/internal/templates/installer.sh:150]: @brentleyjones I forget—will this force an Xcode project change prompt?

Comment (brentleyjones) [reply xcodeproj/internal/templates/installer.sh:150]: I believe it's just `project.xcworkspace`: https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/570175fb14305d44062a15b660d659f2fa4853e7/xcodeproj/internal/templates/installer.sh#L143

**Review by erikkerber — APPROVED** [review]
I think my team would appreciate this for sure.

**Review by luispadron — APPROVED** [review]
Sort order makes sense to me! This looks like a good improvement thanks

File: tools/generator/src/Generator/CreateXcodeProj.swift [hunk 2]
```diff
@@ -4,12 +4,14 @@ extension Generator {
     /// Creates an `XcodeProj` for the given `PBXProj`.
     static func createXcodeProj(
         for pbxProj: PBXProj,
-        sharedData: XCSharedData?
+        sharedData: XCSharedData?,
+        userData: [XCUserData]
```
Comment (brentleyjones) [inline tools/generator/src/Generator/CreateXcodeProj.swift:8]: Until we need this to be an array, we should accept a single value here, and wrap it in an array in the `XcodeProj` call below.
```suggestion
        userData: XCUserData
```

Comment (David-Lam) [reply tools/generator/src/Generator/CreateXcodeProj.swift]: Sounds good! Changed to wrapping `XCUserData` in `XcodeProj` call everywhere https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2151/commits/dbea5e45c39c8826c2951ce1799fc5aac7cc6d84

**Review by brentleyjones — APPROVED** [review]

**Comment by brentleyjones** [comment]
Thanks @David-Lam!

**Pull request merged**